Add new important best practice for variable loops (#1029)
harness/developer-hub: docs/platform/8_Pipelines/best-practices-for-looping-strategies.md

2. Run the Pipeline and monitor the resource consumption for the overall Pipeline.
3. Gradually increase the `maxConcurrency` based on each successive run until you reach a "happy medium" between your run times and resource consumption.

> ### How to use variables representing lists/arrays to loop and repeat for each item

Reviewer: ```suggestion
### What is the best way to loop over items in a list or array using variables?
```

> You can use all Java String class built-in methods on Harness variable expressions.

Reviewer: ```suggestion
Harness variable expressions support all Java String class built-in methods.
```

> With that in mind, you can define a Harness string variable containing a comma-separated list of items represented by other strings. You can even get this values list from inputs at runtime.  

Reviewer: ```suggestion
You can create a Harness string variable containing a comma-separated list of other strings that represent items. At runtime, you can even get this list from the input.   
```

> For example, this Pipeline variable that contains a list of Jira Tickets:

Reviewer: ```suggestion
The following example shows a pipeline variable that contains a list of Jira tickets:
```

```
# Variable: <+pipeline.variables.jiraTickets>
HD-29193,HD-29194,HD-29195

# YAML representation
pipeline:
  identifier: RepeatJiraTickets
  variables:
    - name: jiraTickets
      type: String
      value: HD-29193,HD-29194,HD-29195 // We have added the list of tickets to be comma separated
```

> By using the `split()` method, you can split that variable string into an array of substrings:

Reviewer: ```suggestion
To split the variable into an array of substrings, use the `split()` method:
```

```
<+pipeline.variables.jiraTickets.split(',')>
```

> You can use this expression in your **repeat** Looping Strategy. For example:

Reviewer: ```suggestion
The following example shows how to use the above expression in the `repeat` looping strategy:
```

```
repeat:
  items: <+stage.variables.jiraTickets.split(',')>
  maxConcurrency: 1
```

> Finally, you refer to each item in the loop with the `<+repeat.item>` expression.

Reviewer: ```suggestion
To refer to each item in the loop use the `<+repeat. Item>` expression:
```

![Repeat with split()](./static/best-practices-for-looping-strategies-08.png)

> **Extra:** You can also dynamically create an axis for your Matrix.

Reviewer: ```suggestion
You can also create an axis for your matrix dynamically using the following expression:
```

```
matrix:
  jira: <+stage.variables.jiraTickets.split(',')>
```

> And use the `<+matrix.jira>` expression instead.

Reviewer: To reference this axis use `<+matrix.jira>`.
